qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/results_tab_widget/results_tab_presenter.py

```diff
@@ -107,6 +107,11 @@ def _update_fit_results_view_on_new_fit(self):
         """Update the view of results workspaces based on the current model"""
         workspace_list, function_name = self._get_workspace_list()
         self.view.set_selected_fit_function(function_name)
+
+        current_view = self.view.fit_result_workspaces()
+        not_selected = [key for key in current_view.keys() if not current_view[key][1]]
+        workspace_list = [ws for ws in workspace_list if ws not in not_selected]
```

Comment:
could we not do something like the below? Or is there a reason this is not possible?
```
for ws in reversed(workspace_list):
     if not current_view[ws][1]:
              workspace_list.remove(ws)

```

Reply:
The current view does not have the new ws in it, so we can not rely on current_view[ws] nt failing.
We could add an `if ws in current_view.keys():` but I'm not sure if there is any value in that.

Reply:
ok, just asking as it seemed odd to me to make a list, to then make another list based of a 3rd list. If thats the best way to do it, there is not much else we can do.